src/vm/builtins/builtin_runner/signature.zig

```diff
@@ -78,4 +155,365 @@ pub const SignatureBuiltinRunner = struct {
         _ = self;
         return null;
     }
+
+    pub fn ratio(self: *const Self) ?u32 {
```

Comment:
Here we don't need this one as long as `self.ratio` is public and not private in Zig.